unit_test/test_utils.c

```diff
@@ -0,0 +1,543 @@
+#include "test_utils.h"
+#include <stdio.h>
+#include <stdlib.h>
+
+#define ERROR_HEADER_FORMAT "\033[0;31m[ERROR] %s... "

```

Comment:
```suggestion
#define printf_error_header(s)                               printf("\033[0;31m[ERROR] in %s: %s... \033[0m", __func__, s)
#define printf_error_expected_got(f,exp,got)     printf("\033[0;31m" f "\033[0m", exp, got)
#define printf_error(s)                                             printf("\033[0;31m" s "\033[0m")

```

Reply:
I do not totally agree with this full-macro policy since there is no type-check for macros

Reply:
And what is `__func__`?